packages/plugin-twitter/src/actions/post.ts

```diff
@@ -6,115 +6,67 @@ import {
-            template: tweetTemplate,
-        });
-
-        const tweetContentObject = await generateObject({
-            runtime,
-            context,
-            modelClass: ModelClass.SMALL,
-            schema: TweetSchema,
-            stop: ["\n"],
-        });
+function truncateToCompleteSentence(text: string, maxLength: number): string {
+    if (text.length <= maxLength) {
+        return text;
+    }
 
-        if (!isTweetContent(tweetContentObject.object)) {
-            elizaLogger.error(
-                "Invalid tweet content:",
-                tweetContentObject.object
-            );
-            return;
+    // Attempt to truncate at the last period within the limit
+    const lastPeriodIndex = text.lastIndexOf(".", maxLength - 1);
+    if (lastPeriodIndex !== -1) {
+        const truncatedAtPeriod = text.slice(0, lastPeriodIndex + 1).trim();
+        if (truncatedAtPeriod.length > 0) {
+            return truncatedAtPeriod;
         }
+    }
 
-        const trimmedContent = tweetContentObject.object.text.trim();
-
-        // Skip truncation if TWITTER_PREMIUM is true
-        if (
-            process.env.TWITTER_PREMIUM?.toLowerCase() !== "true" &&
-            trimmedContent.length > 180
-        ) {
-            elizaLogger.warn(
-                `Tweet too long (${trimmedContent.length} chars), truncating...`
-            );
-            return trimmedContent.substring(0, 177) + "...";
+    // If no period, truncate to the nearest whitespace
+    const lastSpaceIndex = text.lastIndexOf(" ", maxLength - 1);
+    if (lastSpaceIndex !== -1) {
+        const truncatedAtSpace = text.slice(0, lastSpaceIndex).trim();
+        if (truncatedAtSpace.length > 0) {
+            return truncatedAtSpace + "...";
         }
-
-        return trimmedContent;
-    } catch (error) {
-        elizaLogger.error("Error composing tweet:", error);
-        throw error;
     }
+
+    // Fallback: Hard truncate and add ellipsis
+    const hardTruncated = text.slice(0, maxLength - 3).trim();
+    return hardTruncated + "...";
 }
 
-async function postTweet(content: string): Promise<boolean> {
+async function postTweet(
+    runtime: IAgentRuntime,
+    content: string
+): Promise<boolean> {
     try {
-        const scraper = new Scraper();
-        const username = process.env.TWITTER_USERNAME;
-        const password = process.env.TWITTER_PASSWORD;
-        const email = process.env.TWITTER_EMAIL;
-        const twitter2faSecret = process.env.TWITTER_2FA_SECRET;
-
-        if (!username || !password) {
-            elizaLogger.error(
-                "Twitter credentials not configured in environment"
-            );
-            return false;
-        }
-
-        // Login with credentials
-        await scraper.login(username, password, email, twitter2faSecret);
-        if (!(await scraper.isLoggedIn())) {
-            elizaLogger.error("Failed to login to Twitter");
+        const twitterClient = runtime.clients.twitter;
```

Comment:
I think if you do this, people who aren’t running the Twitter client might not be able to use this plugin anymore. Maybe you could try using cookies to resolve the issue. (Reference: https://github.com/elizaOS/agent-twitter-client/blob/ffcfed49c35b1130540c45eef95086a9fb6dff8a/src/test-utils.ts#L16)

Reply:
Or an option for both, those who use it along with client-twitter and with out it 🤔

Reply:
Yes, it's definitely an improvement 👍. BTW could you please modify the env.process in this plugin to use runtime.getSetting instead?